packages/web-components/fast-foundation/src/accordion/accordion.ts

```diff
@@ -111,18 +110,28 @@ export class FASTAccordion extends FASTElement {
             item.addEventListener("keydown", this.handleItemKeyDown);
             item.addEventListener("focus", this.handleItemFocus);
         });
+
         if (this.isSingleExpandMode()) {
-            const expandedItem: FASTAccordionItem | null =
-                this.findExpandedItem() ??
-                ((this._accordionItems[0] as unknown) as FASTAccordionItem);
-            expandedItem.setAttribute("aria-disabled", "true");
+            const expandedItem = this.findExpandedItem() as FASTAccordionItem;
+            this.setSingleExpandMode(expandedItem);
         }
     };
 
-    private resetItems(): void {
-        // reset ALL items if we swap to single expand mode (should we reset an expanded item which is disabled?)
-        this.accordionItems.forEach((item: FASTAccordionItem, index: number) => {
-            item.expanded = false;
+    private setSingleExpandMode(expandedItem: Element): void {
+        const currentItems = Array.from(this._accordionItems);
+        this.activeItemIndex = currentItems.indexOf(expandedItem);
+
+        currentItems.forEach((item: FASTAccordionItem, index: number) => {
+            if (this.activeItemIndex === index) {
+                item.expanded = true;
+                item.expandbutton.setAttribute("aria-disabled", "true");
+            } else {
+                item.expanded = false;
+
+                if (!item.hasAttribute("disabled")) {
+                    item.removeAttribute("aria-disabled");
+                }
```

Comment:
Same here, could this be handled with the observable `disabled` for the `accordion-item`?

Reply:
I think the code here is slightly wrong, but as noted above this is different than disabling an item and more about disabling the button of the item to prevent the open item from collapsing in single-expand mode. More here: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/